<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="model"     default="$(find xacro)/xacro '$(find pivot_base)/urdf/orange.xacro'" />
  <arg name="x_pos"     default="0.0"/>
  <arg name="y_pos"     default="0.0"/>
  <arg name="z_pos"     default="0.1"/>
  <arg name="roll_pos"  default="0.0"/>
  <arg name="pitch_pos" default="0.0"/>
  <arg name="yaw_pos"   default="0.0"/>

  <arg name="map_file" default="$(find pivot_base)/maps/mymap_nakaniwa_sim.yaml"/>
  <arg name="test_folder" default="$(find pivot_base)/test"/>
  <arg name="down_scale_factor" default="0.05"/>
  <arg name="robot_radius" default="0.15"/>
  <arg name="odom_topic" default="/odom"/>
  <arg name="scan_topic" default="/scan"/>
  <arg name="vel_linear" default="0.1"/>
  <arg name="vel_theta" default="0.3"/>
  <arg name="angle_tolerance" default="0.1"/>
  <arg name="goal_tolerance" default="0.05"/>
  <arg name="verbose" default="false"/>
  <arg name="open_rviz" default="true"/>

  <include file="$(find pivot_base)/launch/include/pivot_world.launch">
    <arg name="model" value="$(arg model)"/>
    <arg name="x"     value="$(arg x_pos)"/>
    <arg name="y"     value="$(arg y_pos)"/>
    <arg name="z"     value="$(arg z_pos)"/>
    <arg name="roll"  value="$(arg roll_pos)"/>
    <arg name="pitch" value="$(arg pitch_pos)"/>
    <arg name="yaw"   value="$(arg yaw_pos)"/>
  </include>

  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" output="screen"/>
    
  <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
    <param name="use_gui" value="true"/>
    <param name="rate"    value="50"/>
  </node>
  <!--<include file="$(find turtlebot3_bringup)/launch/turtlebot3_remote.launch">
    <arg name="model" value="$(arg model)" />
  </include>-->

  <include file="$(find pivot_base)/launch/include/pivot_amcl.launch">
    <arg name="scan_topic"      value="$(arg scan_topic)"/>
    <arg name="initial_x_pos"   value="$(arg x_pos)"/>
    <arg name="initial_y_pos"   value="$(arg y_pos)"/>
    <arg name="initial_yaw_pos" value="$(arg yaw_pos)"/>
  </include>

  <include file="$(find pivot_base)/launch/include/pivot_planner.launch">
    <arg name="map_file" value="$(arg map_file)"/>
    <arg name="test_folder" value="$(arg test_folder)"/>
    <arg name="down_scale_factor" value="$(arg down_scale_factor)"/>
    <arg name="robot_radius" value="$(arg robot_radius)"/>
    <arg name="odom_topic" value="$(arg odom_topic)"/>
    <arg name="scan_topic" value="$(arg scan_topic)"/>
    <arg name="vel_linear" value="$(arg vel_linear)"/>
    <arg name="vel_theta" value="$(arg vel_theta)"/>
    <arg name="angle_tolerance" value="$(arg angle_tolerance)"/>
    <arg name="goal_tolerance" value="$(arg goal_tolerance)"/>
    <arg name="verbose" value="$(arg verbose)"/>
  </include>

  <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)"/>

  <node if="$(arg open_rviz)" name="rviz" pkg="rviz" type="rviz" required="true" args="-d $(find pivot_base)/rviz/pivot_navigation.rviz"/>
</launch>
